src/ssl.c

```diff
@@ -981,6 +981,140 @@ static int openssl_ssl_ctx_set_cert_verify(lua_State*L)
   return 0;
 }
 
+#if OPENSSL_VERSION_NUMBER >= 0x10002000L
+/***
+set the list of ALPN protocols available to be negotiated
+@function set_alpn_protos
+@tparam[opt] table protos the protocol list
```

Comment:
@tparam table protos the protocol list
@tparam function callback

This api can accept two paramaters, arg 1 is protos table, arg 2 is callback function. One time lua call wrap two OpenSSL APIs, Lua should make that simple.

Reply:
As far as I understood, the protocol list is provided by the client and the selection callback is used on the server side.
I would recommend to stick to the openssl two methods and naming but do not hesitate to indicate if you still want one method.

Reply:
It's ok